### PR TITLE
Spanner DML

### PR DESCRIPTION
**Pull request by jskeet**
This is not ready for submission quite yet, as the public endpoint doesn't have it enabled.

It's all been reviewed already, but another once-over wouldn't hurt...

### CODE REVIEW

**Comment by jskeet** [comment]
I've just thought of an option to make this review simpler: I could create a separate PR just for the codegen, then rebase off that. Then this PR would *only* contain the manual code changes. Would that be better for you?

**Comment by amanda-tarafa** [comment]
Yest, that might be better I think. Thanks!

**Comment by jskeet** [comment]
Done - this is a slightly more sensible PR now :)

**Comment by amanda-tarafa** [comment]
Thanks, looking now!

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void InvalidQuery()
+        {
+            string key = CreateTestRows();
+            using (var connection = _fixture.GetConnection())
+            {
+                // Slightly odd choice here: "OriginalValue" has to be unique, so we end up with the inserted Value being the real original value.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs]: I think this comment is out of place here.

Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs]: Nope, that's deliberate - basically "original value" doesn't mean what one might expect it to.
Happy to reword the comment though.

Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs]: I just don't understand it in this test. It's in the test above and there it makes sense, but in this one?

Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs]: In other tests, OriginalValue is the value the row was created with, and Value is a modified value.

In this case we're *inserting* new rows based on existing rows - so they can't share the same OriginalValue.

Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs]: I think you are referring to the above test, the one in line 189 called InsertWithQuery, and there I get the comment.
But it being here in this test called InvalidQuery, in which you are just using an invalid query to test that it throws, I think makes no sense. I think this was just a copy paste thing?

Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs]: Aha! Yes, sorry. Yup, this was indeed just a C&P fail. I'll remove the comment.

**Pull request merged**